packages/graphiql-plugin-code-exporter/src/graphiql-code-exporter.d.ts

```diff
@@ -0,0 +1,53 @@
+declare module 'graphiql-code-exporter' {
+  import { ComponentType } from 'react';
+  import {
+    GraphQLSchema,
+    OperationTypeNode,
+    OperationDefinitionNode,
+    FragmentDefinitionNode,
+  } from 'graphql';
+
+  type OperationData = {
+    query: string;
+    name: string;
+    displayName: string;
+    type: OperationTypeNode | 'fragment';
+    variableName: string;
+    variables: Record<string, any>;
+    operationDefinition: OperationDefinitionNode | FragmentDefinitionNode;
+    fragmentDependencies: Array<FragmentDefinitionNode>;
+  };
+
+  type GenerateOptions = {
+    serverUrl: string;
+    headers: Record<string, string>;
+    context: Record<string, any>;
+    operationDataList: Array<OperationData>;
+    options: Record<string, boolean>;
+  };
+
+  type Snippet = {
+    language: string;
+    codeMirrorMode: string;
+    name: string;
+    options: Array<{
+      id: string;
+      label: string;
+      initial: boolean;
+    }>;
+    generate: (options: GenerateOptions) => string;
+  };
+
+  export type GraphiQLCodeExporterProps = {
+    query: string;
+    snippets: Array<Snippet>;
+    codeMirrorTheme: string;
+    variables?: string;
+    context?: Record<string, any>;
+    schema?: GraphQLSchema | null | undefined;
+  };
```

Comment:
These are different to the prop definitions that I find in the [source repository](https://github.com/OneGraph/graphiql-code-exporter/blob/ab861a9ab0dbcf8b566d120f10be589055e3e7c2/src/CodeExporter.js#L740-L755):
- `codeMirrorTheme` is optional in the source repo but required here
- `variables` and `context` are required in the source repo but optional here
- A bunch of other props from the source repo are just missing here
Are there particular reasons for deviating from the prop definitions? Could we try to mirror them here instead of just declaring a subset?

Reply:
> codeMirrorTheme is optional in the source repo but required here

With the ref handling for some reason the default props weren't applied. So unless (with the current code) you define the `codeMirrorTheme` explicitly, the whole thing didn't work.

Maybe when I'll remove the wrapper it works, I'll test it.

> variables and context are required in the source repo but optional here

But they are set with `{}` defaults. So actually their types are incorrect and they are optional.

> A bunch of other props from the source repo are just missing here

I didn't find them useful at the time, as I just wanted to make something minimal working. But I can copy the types, yeah.

Reply:
Also the props differ from README to code 😆 
https://github.com/OneGraph/graphiql-code-exporter#props